games-emulation/retroarch/retroarch-1.7.1.recipe

```diff
@@ -0,0 +1,91 @@
+SUMMARY="The reference frontend of libretro"
+DESCRIPTION="RetroArch is a frontend for emulators, game engines and media \
+players. It enables you to run classic games on a wide range of computers and \
+consoles through its slick graphical interface. It has advanced features like \
+shaders, netplay, rewinding, next-frame response times, and more!"
+HOMEPAGE="https://libretro.com/"
+COPYRIGHT="2010-2018 The RetroArch Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/libretro/RetroArch/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="e9189c7da6b2ca0142528eca14f26104ee322f052f8609203d90b60938d8d3f3"
+SOURCE_FILENAME="retroarch-$portVersion.tar.gz"
+SOURCE_DIR="RetroArch-$portVersion"
+
+srcGitRev_2="505ad6205e5fd8a155daad02ab39953a0b15b0a9"
+SOURCE_URI_2="https://github.com/libretro/retroarch-assets/archive/$srcGitRev_2.tar.gz"
+CHECKSUM_SHA256_2="de37772b00b755b80aafe5d4df0713c0fe6b0e5a79017d94e43f279e798aa85d"
+SOURCE_FILENAME_2="retroarch-assets-$srcGitRev_2.tar.gz"
```

Comment:
Oops, you ate the `-$portVersion` :)
Well, this is OK as is, but what I'd recomment is:
```diff
-SOURCE_FILENAME_2="retroarch-assets-$srcGitRev_2.tar.gz"
+SOURCE_FILENAME_2="retroarch-assets-$portVersion-$srcGitRev_2.tar.gz"
```
I agree that this is not required, but it is handy if one wants to manually remove an outdated tarball, since it is much easier fo figure out which tarballs are outdated and which are not. And this (currently) only adds 6 chars to the name of the tarball.

Everything else looks OK, except that I forgot to point that we have recently chosen to recommend putting `GLOBAL_WRITABLE_FILES` and `USER_SETTINGS_FILES` **_before_** `PROVIDES`:
```diff
 SECONDARY_ARCHITECTURES="x86"
+
+GLOBAL_WRITABLE_FILES="
+	$relativeSettingsDir/retroarch.cfg auto-merge
+	"
+USER_SETTINGS_FILES="
+	$relativeSettingsDir/retroarch.cfg template \
+		$relativeDataDir/retroarch/retroarch.cfg.default
+	"
+
 PROVIDES="
```
We have several recipes which have them after `BUILD_PREREQUIRES`, but we have many more which have it before `PROVIDES`. So now, everytime we can "fix" this without too much git noise, we try to do so :)

[HaikuPorter-Guidelines#ordering](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering)

Reply:
This topic a not easy, and I'm not even sure I already understood it correctly :)
* [`GLOBAL_WRITABLE_FILES`](https://github.com/haikuports/haikuports/wiki/HaikuPorter-BuildRecipes#global_writable_files) is used to let the system know that a package may create files in `/system/settings/` (or any subdirectory of it.) Mentioning a config file here does not mean the package comes with that file. It only means the package may create that config file.
* [`USER_SETTINGS_FILES`](https://github.com/haikuports/haikuports/wiki/HaikuPorter-BuildRecipes#user_settings_files) is used when the package comes with a config file, and also allows to specify the default config. BTW, if the `template` keyword is used then when a package is installed for the very first time, the config should be identical to the template, which is generally in a subdirectory of `$dataDir/` (i.e. `/system/data/<subdir>/`).